modules/apps/27-interchain-accounts/controller/types/msgs.go

```diff
@@ -15,10 +22,27 @@ func NewMsgRegisterAccount(connectionID, owner, version string) *MsgRegisterAcco
 
 // ValidateBasic implements sdk.Msg
 func (msg MsgRegisterAccount) ValidateBasic() error {
+	if err := host.ConnectionIdentifierValidator(msg.ConnectionId); err != nil {
+		return sdkerrors.Wrap(err, "invalid connection ID")
+	}
+
+	if strings.TrimSpace(msg.Owner) == "" {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, "owner address cannot be empty")
+	}
+
+	if _, err := sdk.AccAddressFromBech32(msg.Owner); err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "failed to parse owner address: %s", msg.Owner)
+	}
```

Comment:
do we also need a validation on the version?

Reply:
I don't think we can since this version may be wrapped with many different versions (ics29 and so forth). There's no guarantee that the provided version is an ics27 version at the top level